README.md

```diff
@@ -25,25 +25,64 @@ For an example see the [server.go](https://github.com/openfaas/faas-netes/blob/m
 I.e.:
 
 ```go
-	bootstrapHandlers := bootTypes.FaaSHandlers{
-		FunctionProxy:  handlers.MakeProxy(),
-		DeleteHandler:  handlers.MakeDeleteHandler(clientset),
-		DeployHandler:  handlers.MakeDeployHandler(clientset),
-		FunctionReader: handlers.MakeFunctionReader(clientset),
-		ReplicaReader:  handlers.MakeReplicaReader(clientset),
-		ReplicaUpdater: handlers.MakeReplicaUpdater(clientset),
-		InfoHandler:    handlers.MakeInfoHandler(),
-	}
-
-	var port int
-	port = 8080
-	bootstrapConfig := bootTypes.FaaSConfig{
-		ReadTimeout:  time.Second * 8,
-		WriteTimeout: time.Second * 8,
-		TCPPort:      &port,
-	}
-
-	bootstrap.Serve(&bootstrapHandlers, &bootstrapConfig)
+import (
+	bootTypes "github.com/openfaas/faas-provider/types/v1"
+	bootstrap "github.com/openfaas/faas-provider/bootstrap/v1"
+)
+
+bootstrapHandlers := bootTypes.FaaSHandlers{
+    FunctionProxy:  handlers.MakeProxy(),
+    DeleteHandler:  handlers.MakeDeleteHandler(clientset),
+    DeployHandler:  handlers.MakeDeployHandler(clientset),
+    FunctionReader: handlers.MakeFunctionReader(clientset),
+    ReplicaReader:  handlers.MakeReplicaReader(clientset),
+    ReplicaUpdater: handlers.MakeReplicaUpdater(clientset),
+    InfoHandler:    handlers.MakeInfoHandler(),
+}
+
+var port int
+port = 8080
+bootstrapConfig := bootTypes.FaaSConfig{
+    ReadTimeout:  time.Second * 8,
+    WriteTimeout: time.Second * 8,
+    TCPPort:      &port,
+}
+
+bootstrap.Serve(&bootstrapHandlers, &bootstrapConfig)
+```
+
+### Upgrade to OpenFaaS provider v2 packages
+
+Example of v2 provider bootstrap:
+
+```go
+import bootstrap "github.com/openfaas/faas-provider/bootstrap/v2"
+
+env := bootstrap.EnvReader{}
+
+config := bootstrap.ApiServerConfig{
+	ReadTimeout:  env.ParseDuration("read_timeout", time.Second*10),
```

Comment:
How is error handling covered for `env.ParseDuration`? Just hide / swallow and use the default?

Reply:
the EnvReader code in this PR:

```go

func (*EnvReader) ParseDuration(key string, fallback time.Duration) time.Duration {
	val := os.Getenv(key)
	if len(val) > 0 {
		parsedVal, parseErr := strconv.Atoi(val)
		if parseErr == nil && parsedVal >= 0 {
			return time.Duration(parsedVal) * time.Second
		}
	}

	duration, durationErr := time.ParseDuration(val)
	if durationErr != nil {
		return fallback
	}

	return duration
}
```